### PR TITLE
Refactor/makei

### LINKED ISSUE #23: Detected dependencies should be qualified to the project directory
We now support multiple directory builds.  So the included files and other modules and service programs will likely be in a different directory.
For example when building ART300.MODULE in  https://github.com/edmundreinhardt/bob-recursive-example
the .evfevent that is generated is

TIMESTAMP  0 20210920172044
PROCESSOR  0 000 1
FILEID     0 001 000000 076 /home/REINHARD/bob-recursive-example/QRPGLESRC/ART300-Function_Article.RPGLE 20210918015716 0
FILEID     0 002 000005 063 /home/REINHARD/bob-recursive-example/qprotosrc/article.rpgleinc 20210918015716 0

But the current algorithm doesn't take the relative path up to the project root which should be:
- QRPGLESRC/ART300-Function_Article.RPGLE
- qprotosrc/article.rpgleinc

instead it generates 
ART300.MODULE: ARTICLE.RPGLEINC
ART300.MODULE: ART300-FUNCTION_ARTICLE.RPGLE
ART300.MODULE:   ARTICLE1.FILE

At make time, there is no target for these dependencies and so it fails.

### LINKED ISSUE #53: Init parameter
I didn’t see any docs about what the initial `iproj.json` is.

I think it’d be nice for `imake` to have an `init` parameter to generate two initial files:

* `iproj.json`
* root `Rules.mk` with `SUBDIRS` already setup.

### PR DESCRIPTION
**Pull request by TongkunZhang**
- Rewrite `makei` in Python for better performance and extensibility
- Refactored the `makei` cli
```
$ makei -h
usage: makei [-h] command ...

optional arguments:
  -h, --help   show this help message and exit

These are common makei commands:
  command
    init       set up a new or existing project
    compile (c)
               compile a single file
    build (b)  build the whole project

```
- add `makei init` subprogram to set up a project Fixes #53 
@edmundreinhardt Please comment on which parameters to keep.

- allow compile multiple files for `makei compile` command [Updated: Feb 24]

- Fixes #23 

TODOs:

- [x] Allow override of BOB_PATH for developing

### CODE REVIEW

File: makei [hunk 1]
```diff
@@ -1,226 +1,201 @@
...
+        help='compile a single file')
+    compile_target_group = compile_parser.add_mutually_exclusive_group(required=True)
+    compile_target_group.add_argument(
+        '-f',
+        '--file',
+        help='file to compile',
+        metavar='<filename>')
+    compile_target_group.add_argument(
+        '--files',
+        help='files to compile',
```
Comment (edmundreinhardt) [inline makei:99]: what is path separation (:)

File: scripts/init_project.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        input_str.strip().lower() == "y" or \
+        input_str.strip() == ""
+
+
+def init_cancelled():
+    print(colored('\nInit cancelled!', Colors.WARNING))
+    sys.exit(0)
+
+
+def prompt(description, default_vaule) -> str:
```
Comment (edmundreinhardt) [inline scripts/init_project.py:144]: sp value

File: scripts/init_project.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        'This utility will walk you through creating a project.',
+        'It only covers some common items.',
+        '',
+        'Press ^C at any time to quit.',
+    ]))
+
+    proj_spec = ProjSpec()
+
+    iproj_json_path = Path.cwd() / 'iproj.json'
+    ibmi_json_path = Path.cwd() / '.ibmi.json'
```
Comment (edmundreinhardt) [inline scripts/init_project.py:173]: Should only generate .ibmi.json if a non-default tgtccsid

File: scripts/utils.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            print(colored(f"{varName} must be defined first in the environment variable.", Colors.FAIL))
+            exit(1)
+    else:
+        return varName
+
+def read_iproj_json(iproj_json_path):
+    try:
+        with iproj_json_path.open() as f:
+            iproj_json = json.load(f)
+            objlib = parse_placeholder(iproj_json["objlib"]) if "objlib" in iproj_json else ""
```
Comment (edmundreinhardt) [inline scripts/utils.py:70]: are we pulling in defaults if these are not specified?  i.e. *CURLIB and *CRTDFT

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]

File: scripts/build.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                    f"OBJPATH_{subdir.absolute()} := {objlib_to_path(dir_var_map[subdir][0])}\n")
+
+            for rules_mk in rules_mks:
+                with rules_mk.open('r') as r:
+                    ls = r.readlines()
+                    for l in ls:
+                        l = l.rstrip()
+                        if l and not l.startswith("#") and not "=" in l and not l.startswith((' ', '\t')):
+                            f.write(
+                                f"{l.split(':')[0]}_d := {rules_mk.parents[0].absolute()}\n")
```
[the remark this answers is not in the document]
Comment (TongkunZhang) [reply scripts/build.py:112]: Tracking [here](https://github.com/IBM/ibmi-bob/issues/66)

File: scripts/init_project.py (hunk 2 above)
Comment (edmundreinhardt) [reply scripts/init_project.py:158]: default_vaule should be spelled default_value

Comment (TongkunZhang) [reply scripts/init_project.py]: Thanks

**Pull request merged**